### PR TITLE
Fix: Query Compilation when using References (SIRI-288)

### PR DESCRIPTION
**Pull request by jakobvogel**
> https://scireum.myjetbrains.com/youtrack/issue/SIRI-288

- Protects query compilation against exceptions

### CODE REVIEW

File: src/main/java/sirius/db/mixing/query/QueryCompiler.java [hunk 1]
```diff
@@ -710,7 +711,21 @@ protected Object compileStringValue(Property property, String value) {
...
+        }
+
+        try {
+            return property.transformValue(Value.of(value));
+        } catch (Exception e) {
+            // second line of defence: should the transformation fail for whatever reason, we pass the original string
+            // as is, leaving it to the database to come up with no results if searching for an invalid value (such as a
+            // string in a numeric field)
+            return Value.of(value);
+        }
```
Comment (jakobvogel) [inline src/main/java/sirius/db/mixing/query/QueryCompiler.java:728]: Do we want to do this? Originally, when looking for a string in an integer field (for example), there was an error that the type is illegal. With the try/catch, we leave it to the database to handle that case.

Comment (andyHa) [reply src/main/java/sirius/db/mixing/query/QueryCompiler.java:728]: I'd keep this in place but also add "Exceptions.ignore(e)" this signals that the ex is deliberately ignore and also - for the desperate - there is a logger to enable und view these exception

Comment (jakobvogel) [reply src/main/java/sirius/db/mixing/query/QueryCompiler.java]: Done 👍

**Review by andyHa — APPROVED** [review]

File: src/main/java/sirius/db/mixing/query/QueryCompiler.java [hunk 2]
```diff
@@ -710,7 +712,22 @@ protected Object compileStringValue(Property property, String value) {
             }
         }
 
-        return property.transformValue(Value.of(value));
+        // first line of defence: reference types are not processed as transformation triggers a lookup in the
```
Comment (fhaScireum) [inline src/main/java/sirius/db/mixing/query/QueryCompiler.java:715]:
```suggestion
        // first line of defence: reference types are not processed as transformation triggers at lookup in the
```

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/db/mixing/query/QueryCompiler.java (hunk 2 above)
Comment (jakobvogel) [reply src/main/java/sirius/db/mixing/query/QueryCompiler.java:715]: Nope. 🙂 The transformation _here_ (from `String` to `Value`) triggers the lookup in the _other_ database (where the reference in question points to), not the other way around.

**Review by sabieber — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**